lib/CrudService.js

```diff
@@ -340,6 +340,38 @@ class CrudService {
     return writeOpResult.value
   }
 
+  async upsertMany(context, filterUpdateCommands) {
+    context.log.debug(filterUpdateCommands, 'upsertMany operation requested')
+    assert(filterUpdateCommands.length > 0, 'At least one element is required')
+
+    const operations = []
+    for (let i = 0; i < filterUpdateCommands.length; i++) {
+      const { _id, commands } = filterUpdateCommands[i]
+
+      assertCommands(commands)
+
+      const searchQuery = { _id }
+
+      commands.$set.updaterId = context.userId
+      commands.$set.updatedAt = context.now
+
+      operations.push({
+        updateOne: {
+          filter: searchQuery,
```

Comment:
What happens in case `_id` is empty?

Reply:
It would perform the filter operation with `{ _id: undefined }` which of course leads to 0 results and thus in the end it would perform an insert of the record.